snuba/datasets/processors/search_issues_processor.py

```diff
@@ -0,0 +1,106 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, Tuple, TypedDict
+
+from snuba import environment
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.datasets.events_format import EventTooOld, enforce_retention
+from snuba.datasets.processors import DatasetMessageProcessor
+from snuba.processor import InsertBatch, ProcessedMessage
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+metrics = MetricsWrapper(environment.metrics, "search_issues.processor")
+
+
+class SearchIssueData(TypedDict, total=False):
+    fingerprint: Sequence[str]
+    occurrence_id: str
+    occurrence_type_id: int
+
+    # occurrence-related
+    event_id: Optional[str]
+    trace_id: Optional[str]
+    platform: str
+    environment: Optional[str]
+    release: Optional[str]
+    dist: Optional[str]
+    receive_timestamp: float
+    client_timestamp: float
+    tags: Mapping[str, Any]
+    user: Mapping[str, Any]  # user, user_hash, user_id, user_name, user_email
+    ip_address: str
+    sdk: Mapping[str, Any]  # sdk_name, sdk_version
+    contexts: Mapping[str, Any]
+    request: Mapping[str, Any]  # http_method, http_referer
+
+
+class SearchIssueEvent(TypedDict):
+    # meta
+    retention_days: int
+
+    # issue-related
+    organization_id: int
+    project_id: int
+    group_ids: Sequence[int]
+    search_title: str
+    detection_timestamp: float
+
+    data: SearchIssueData
+
+
+class SearchIssuesMessageProcessor(DatasetMessageProcessor):
+    def process_message(
+        self, message: Tuple[int, str, SearchIssueEvent], metadata: KafkaMessageMetadata
+    ) -> Optional[ProcessedMessage]:
+        if not (isinstance(message, (list, tuple)) and len(message) >= 2):
+            return None
+
+        # TODO: validate version
+        version = message[0]
+        if not version:
+            return None
+
+        type_, event = message[1:3]
+        if type_ != "insert":
+            return None
```

Comment:
Having a type field in the processor was only specific to errors as they have replacers. Do you expect to implement replacement to mutate the table ?

Reply:
For the short term (month or so) we don't, but I think we will have to later down the line.